### PR TITLE
docs(table): added docs around z-index conflicts with use of sticky columns/headers

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/4646

Is this too wordy? I imagine we'll want to add something like this to the react docs, too.

### CODE REVIEW

File: src/patternfly/components/Table/examples/Table.md [hunk 1]
```diff
@@ -2970,6 +2970,8 @@ Add a [width modifier](#width-modifiers) to `thead th` to limit string length or
...
 
+Sticky table headers and columns have a higher z-index than the z-index used for menus (dropdown, select, etc). The intent is that the contents of a scrollable table will scroll under the sticky header/column, including any expanded menus. However, there may be use cases where a menu needs to appear on top of a sticky header/column, such as a menu placed inside of a toolbar above a table with a sticky header. In these cases it's advised to append the menu to an element outside of the table (e.g., the  <code>body</code> element) so that the menu has a higher stacking context than - and can appear on top of - sticky header/columns as well as appear outside of any scrollable content in the table. Typically this is achieved via the <code>menuAppendTo</code> prop in non-composable react components. Another approach may be to manipulate the z-index of the menu manually. In the case where the menu is outside of the table, another approach may be to assign the entire table a z-index lower than the z-index of the menu, creating a lower stacking context for the table compared to the menu outside of the table, preserving the stacking context of the elements inside of the table.
```
Comment (srambach) [inline src/patternfly/components/Table/examples/Table.md:2973]: What about breaking up the paragraph a bit with a list of the options? 
I'd also add "Note:" and set off the z-index and a couple other tweaks. IDK if the first option should be called "recommended" ?

Note: Sticky table headers and columns have a higher `z-index` than the `z-index` used for menus (dropdown, select, etc). The intent is that the contents of a scrollable table will scroll under the sticky header/column, including any expanded menus. However, there may be use cases where a menu needs to appear on top of a sticky header/column, such as a menu placed inside of a toolbar above a table with a sticky header. 
There are a few ways this can be handled:
- Use the <code>menuAppendTo</code> prop in non-composable react components to append the menu to an element outside of the table (e.g., the <code>body</code> element) so that the menu has a higher stacking context than - and can appear on top of - sticky header/columns as well as appear outside of any scrollable content in the table. (Recommended) 
- Manipulate the z-index of the menu manually.
- In the case where the menu is outside of the table, another approach may be to assign the entire table a z-index lower than the `z-index` of the menu. This creates a lower stacking context for the table compared to the menu outside of the table, while preserving the stacking context of the elements inside of the table.

**Review by mattnolting — APPROVED** [review]
Love it, LPTM 🚢
